File: ru.kmorozov.api.onedrive/src/test/java/ru/kmorozov/api/onedrive/test/ReaderTest.java
package ru.kmorozov.api.onedrive.test;

import ru.kmorozov.onedrive.client.OneDriveItem;
import ru.kmorozov.onedrive.client.OneDriveProvider;
import ru.kmorozov.onedrive.client.authoriser.AuthorisationProvider;
import ru.kmorozov.onedrive.client.walker.OneDriveWalkers;
import org.hamcrest.CoreMatchers;
import org.hamcrest.MatcherAssert;
import org.junit.Before;
import org.junit.Test;

import java.io.File;
import java.io.IOException;

/**
 * Created by sbt-morozov-kv on 13.03.2017.
 */
public class ReaderTest {

    private OneDriveProvider api;

    @Before
    public void initApi() throws IOException {
        File file = new File(this.getClass().getClassLoader().getResource("onedrive.key").getFile());

        AuthorisationProvider authoriser = AuthorisationProvider.FACTORY.create(file.toPath(), null, null);
        this.api = OneDriveProvider.FACTORY.readOnlyApi(authoriser);
    }

    @Test
    public void rootTest() throws IOException {
        OneDriveItem root = this.api.getRoot();

        MatcherAssert.assertThat(root, CoreMatchers.is(CoreMatchers.notNullValue()));

        OneDriveItem[] children = this.api.getChildren(root);
        MatcherAssert.assertThat(children, CoreMatchers.is(CoreMatchers.notNullValue()));
    }

    @Test
    public void walkTest() throws IOException {
        OneDriveWalkers.walk(this.api, 3).forEach(oneDriveItem -> System.out.println(oneDriveItem.getName()));
    }
}

File: ru.kmorozov.api.onedrive/src/test/java/ru/kmorozov/api/onedrive/test/ReaderTest.kt
package ru.kmorozov.api.onedrive.test

import ru.kmorozov.onedrive.client.OneDriveProvider
import ru.kmorozov.onedrive.client.OneDriveProvider.FACTORY
import ru.kmorozov.onedrive.client.authoriser.AuthorisationProvider
import ru.kmorozov.onedrive.client.walker.OneDriveWalkers
import org.hamcrest.CoreMatchers
import org.hamcrest.MatcherAssert
import org.junit.Before
import org.junit.Test

import java.io.File
import java.io.IOException

/**
 * Created by sbt-morozov-kv on 13.03.2017.
 */
class ReaderTest {

    private lateinit var api: OneDriveProvider

    @Before
    @Throws(IOException::class)
    fun initApi() {
        val file = File(javaClass.classLoader.getResource("onedrive.key")!!.file)

        val authoriser = AuthorisationProvider.FACTORY.create(file.toPath(), "", "")
        api = FACTORY.readOnlyApi(authoriser)
    }

    @Test
    @Throws(IOException::class)
    fun rootTest() {
        val root = api.root

        MatcherAssert.assertThat(root, CoreMatchers.`is`(CoreMatchers.notNullValue()))

        val children = api.getChildren(root!!)
        MatcherAssert.assertThat(children, CoreMatchers.`is`(CoreMatchers.notNullValue()))
    }

    @Test
    @Throws(IOException::class)
    fun walkTest() {
        OneDriveWalkers.walk(api, 3).forEach { oneDriveItem -> println(oneDriveItem.name) }
    }
}
